packages/esm-ward-app/src/hooks/useInpatientAdmissionByPatients.ts

```diff
@@ -0,0 +1,24 @@
+import { restBaseUrl, useOpenmrsFetchAll } from '@openmrs/esm-framework';
+import { type InpatientAdmission } from '../types';
+
+/**
+ * fetches a list of inpatient admissions (in any location) for the given patients
+ */
+export function useInpatientAdmissionByPatients(patientUuids: string[]) {
+  // prettier-ignore
+  const customRepresentation =
+    'custom:(visit,' +
+    'patient:(uuid,identifiers:(uuid,display,identifier,identifierType),voided,' +
+    'person:(uuid,display,gender,age,birthdate,birthtime,preferredName,preferredAddress,dead,deathDate)),' +
+    'encounterAssigningToCurrentInpatientLocation:(encounterDatetime),' +
+    'currentInpatientRequest:(dispositionLocation,dispositionType,disposition:(uuid,display),dispositionEncounter:(uuid,display),dispositionObsGroup:(uuid,display),visit:(uuid),patient:(uuid)),' +
+    'firstAdmissionOrTransferEncounter:(encounterDatetime),' +
+    'currentInpatientLocation' + 
+    ')';
+
+  return useOpenmrsFetchAll<InpatientAdmission>(
```

Comment:
So, one small thing, but given the very long custom representation here, there's going to be a limited number of patients we can support. Specifically, for GET requests, Chrome is, I think, limited to 2048 characters for the whole URL, which means that we need to somehow paginate things if there are more than say 36 patients in the `patientUuids` array.  2048 (total) - 36^2 (36 UUIDs) - 523 (custom represenation) - 37 (everything else) leaves ~155 characters for the `restBaseUrl`. Since that's 17 characters, we might be able to squeeze the number of patients up to 39.